### PR TITLE
upgrade pytorch and pytorch-lightning requirements and some fixes

### PR DESCRIPTION
**Pull request by kashif**
*Issue #, if available:*

*Description of changes:*

* updated pytorch requirement to >= 1.9 and pytorch-lightning requirement and API
* use torch inference mode during inference which also disables version counting and metadata tracking
* use device in deepAR if available for inference
* update requirements to torch 1.9 and lightning 1.4
* upgraded torch related pips for tests
* some documentation fixes

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

**Please tag this pr with at least one of these labels to make our release process faster:** BREAKING, new feature, bug fix, other change, dev setup

### CODE REVIEW

File: src/gluonts/torch/model/estimator.py [hunk 1]
```diff
@@ -154,7 +154,7 @@ def train_model(
 
         trainer.fit(
             model=training_network,
-            train_dataloader=training_data_loader,
+            train_dataloaders=training_data_loader,
```
Comment (lostella) [inline src/gluonts/torch/model/estimator.py:157]: This used to be different in PL 1.3.1, see https://pytorch-lightning.readthedocs.io/en/1.3.1/api/pytorch_lightning.trainer.trainer.html#pytorch_lightning.trainer.trainer.Trainer.fit

So apparently PL 1.4.0 introduced a breaking change, see https://pytorch-lightning.readthedocs.io/en/1.4.0/api/pytorch_lightning.trainer.trainer.html#pytorch_lightning.trainer.trainer.Trainer.fit

We should probably update the requirement on PL here, too? https://github.com/awslabs/gluon-ts/blob/master/requirements/requirements-pytorch.txt

File: src/gluonts/torch/model/predictor.py [hunk 2]
```diff
@@ -78,7 +78,7 @@ def predict(
 
         self.prediction_net.eval()
 
-        with torch.no_grad():
+        with torch.inference_mode():
```
Comment (lostella) [inline src/gluonts/torch/model/predictor.py:81]: Looks like this was introduced in torch==1.9.0, at least from the documentation, so this probably requires bumping the requirement? https://github.com/awslabs/gluon-ts/blob/master/requirements/requirements-pytorch.txt

File: .github/workflows/tests.yml [hunk 3]
```diff
@@ -9,7 +9,7 @@ jobs:
       max-parallel: 4
       fail-fast: false
       matrix:
-        python-version: [3.6, 3.7, 3.8]
+        python-version: [3.6, 3.7, 3.8, 3.9]
```
Comment (lostella) [inline .github/workflows/tests.yml:12]: Maybe we can enable testing on 3.9 as part of a separate PR @jaheba ?

**Review by lostella — CHANGES_REQUESTED** [review]
Looks good, I have left some comments inline.

It looks like this PR is really about bumping the versions of the adopted frameworks, so maybe the PR title should reflect that?

**Comment by kashif** [comment]
thanks @lostella I'll get it fixed!

File: .github/workflows/tests.yml (hunk 3 above)
Comment (lostella) [reply .github/workflows/tests.yml:12]: Also installing autogluon on python 3.9 appears to be giving issues in the test workflow @kashif so one more reason to split this up :-)

**Comment by kashif** [comment]
@lostella will be ready for another review. Thanks!

**Comment by kashif** [comment]
@jaheba can you kindly merge this before your fixes #1786? Else I might get some conflicts... thanks!

**Comment by jaheba** [comment]
> @jaheba can you kindly merge this before your fixes #1786? Else I might get some conflicts... thanks!

Can do that.

File: .github/workflows/docs.yml [hunk 4]
```diff
@@ -16,7 +16,7 @@ jobs:
 
     - name: Install dependencies
       run: |
-        python -m pip install pip==20.2
+        python -m pip install pip
```
Comment (jaheba) [inline .github/workflows/docs.yml:19]: @kashif Can you revert these changes so we don't get a conflict in #1786?

Thanks!

Comment (jaheba) [reply .github/workflows/docs.yml:19]:
```suggestion
        python -m pip install pip==20.2
```

File: .github/workflows/flake8.yml [hunk 5]
```diff
@@ -14,7 +14,7 @@ jobs:
         python-version: 3.8
     - name: Install dependencies
       run: |
-        python -m pip install pip==20.2
+        python -m pip install pip
```
Comment (jaheba) [inline .github/workflows/flake8.yml:17]:
```suggestion
        python -m pip install pip==20.2
```

File: .github/workflows/mxnet_nightly.yml [hunk 6]
```diff
@@ -24,10 +24,10 @@ jobs:
         python-version: ${{ matrix.python-version }}
     - name: Install dependencies
       run: |
-        python -m pip install pip==20.2
+        python -m pip install pip
```
Comment (jaheba) [inline .github/workflows/mxnet_nightly.yml:27]:
```suggestion
        python -m pip install pip==20.2
```

File: .github/workflows/style_type_checks.yml [hunk 7]
```diff
@@ -14,7 +14,7 @@ jobs:
         python-version: 3.7
     - name: Install dependencies
       run: |
-        python -m pip install pip==20.2
+        python -m pip install pip
```
Comment (jaheba) [inline .github/workflows/style_type_checks.yml:17]:
```suggestion
        python -m pip install pip==20.2
```

File: .github/workflows/test_release_unix_nightly.yml [hunk 8]
```diff
@@ -23,10 +23,10 @@ jobs:
       run: |
         git clone https://github.com/awslabs/gluon-ts --branch $(curl https://api.github.com/repos/awslabs/gluon-ts/releases/latest | grep tag_name | cut -d : -f 2,3 | tr -d \"\ | tr -d \,\ )
         cd gluon-ts
-        python -m pip install pip==20.2
+        python -m pip install pip
```
Comment (jaheba) [inline .github/workflows/test_release_unix_nightly.yml:26]:
```suggestion
        python -m pip install pip==20.2
```

**Review by lostella — DISMISSED** [review]
What @jaheba said, otherwise looks good to me, thank you @kashif!

File: .github/workflows/test_release_win32_nightly.yml [hunk 9]
```diff
@@ -25,10 +25,10 @@ jobs:
         $tmp=$tmp.psobject.properties.value.trim() 
         git clone https://github.com/awslabs/gluon-ts --branch $tmp
         cd gluon-ts
-        python -m pip install pip==20.2
+        python -m pip install pip
```
Comment (jaheba) [inline .github/workflows/test_release_win32_nightly.yml:28]:
```suggestion
        python -m pip install pip==20.2
```

File: .github/workflows/tests.yml [hunk 10]
```diff
@@ -28,9 +28,9 @@ jobs:
       run: pip install mxnet~=1.7.0
     - name: Install other dependencies
       run: |
-        python -m pip install pip==20.2
-        pip install torch==1.8.1+cpu --no-cache-dir -f https://download.pytorch.org/whl/torch_stable.html
-        pip install pytorch-lightning~=1.3.0
+        python -m pip install pip
```
Comment (jaheba) [inline .github/workflows/tests.yml:31]:
```suggestion
        python -m pip install pip==20.2
```

**Review by lostella — APPROVED** [review]

**Comment by kashif** [comment]
@lostella @jaheba sorry missed it... was in a call thanks for doing this!
